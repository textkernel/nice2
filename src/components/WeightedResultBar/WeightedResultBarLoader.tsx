import * as React from 'react';
import { bem } from '../../utils';
import { ContentPlaceholder } from '../ContentPlaceholder';
import { ProgressBar } from '../ProgressBar';
import styles from './WeightedResultBar.scss';

const { block } = bem('WeightedResultBar', styles);

export const WeightedResultBarLoader: React.FC<React.HTMLAttributes<HTMLDivElement>> = (props) => {
    const [percentageToShow, setPercentageToShow] = React.useState(0);

    // simulate prop change on ProgressBar to get a little animation going
    React.useEffect(() => {
        setTimeout(() => {
            setPercentageToShow(100);
        }, 0);
    }, []);

    return (
        <div {...props} {...block(props)}>
            <ContentPlaceholder height={17} width={Math.floor(Math.random() * 60) + 25} />
            <ProgressBar percentage={percentageToShow} context="neutral" small />
        </div>
    );
};

WeightedResultBarLoader.displayName = 'WeightedResultBarLoader';
