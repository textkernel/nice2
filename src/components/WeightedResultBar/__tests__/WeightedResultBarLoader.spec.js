import React from 'react';
import toJson from 'enzyme-to-json';
import { WeightedResultBarLoader } from '../WeightedResultBarLoader';

describe('WeightedResultBarLoader', () => {
    it('should render correctly in loading state', () => {
        const wrapper = mount(
            <div>
                <WeightedResultBarLoader />
                <WeightedResultBarLoader />
            </div>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
