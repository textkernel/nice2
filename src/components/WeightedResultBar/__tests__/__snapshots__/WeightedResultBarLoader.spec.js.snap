// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeightedResultBarLoader should render correctly in loading state 1`] = `
<div>
  <WeightedResultBarLoader>
    <div
      className="WeightedResultBar"
    >
      <ContentPlaceholder
        duration={1}
        height={18}
        width={72}
        withoutMargin={false}
      >
        <div
          className="ContentPlaceholder"
          style={
            Object {
              "animationDuration": "1s",
              "height": 18,
            }
          }
        >
           
          <div
            className="ContentPlaceholder__mask"
            style={
              Object {
                "width": "28%",
              }
            }
          />
        </div>
      </ContentPlaceholder>
      <ProgressBar
        animated={false}
        context="neutral"
        hidden={false}
        percentage={100}
        small={true}
      >
        <div
          aria-hidden={false}
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={100}
          className="ProgressBar ProgressBar--small"
          role="progressbar"
        >
          <div
            className="ProgressBar__fill ProgressBar__fill--context_neutral ProgressBar__fill--small"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </ProgressBar>
    </div>
  </WeightedResultBarLoader>
  <WeightedResultBarLoader>
    <div
      className="WeightedResultBar"
    >
      <ContentPlaceholder
        duration={1}
        height={18}
        width={64}
        withoutMargin={false}
      >
        <div
          className="ContentPlaceholder"
          style={
            Object {
              "animationDuration": "1s",
              "height": 18,
            }
          }
        >
           
          <div
            className="ContentPlaceholder__mask"
            style={
              Object {
                "width": "36%",
              }
            }
          />
        </div>
      </ContentPlaceholder>
      <ProgressBar
        animated={false}
        context="neutral"
        hidden={false}
        percentage={100}
        small={true}
      >
        <div
          aria-hidden={false}
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={100}
          className="ProgressBar ProgressBar--small"
          role="progressbar"
        >
          <div
            className="ProgressBar__fill ProgressBar__fill--context_neutral ProgressBar__fill--small"
            style={
              Object {
                "width": "100%",
              }
            }
          />
        </div>
      </ProgressBar>
    </div>
  </WeightedResultBarLoader>
</div>
`;
